packages/@blockprotocol/core/src/core-block-handler.ts

```diff
@@ -0,0 +1,55 @@
+import { CoreHandler } from "./core-handler";
+import { EmbedderInitMessage, Message } from "./types";
+
+/**
+ * Implements the Block Protocol Core Specification for blocks.
+ */
+export class CoreBlockHandler extends CoreHandler {
+  constructor({ element }: { element: HTMLElement }) {
+    super({ element, sourceType: "block" });
+    this.initialize();
+  }
+
+  /**
+   * Send the init message, to which an initResponse is expected from the embedder.
+   * The response will be processed in {@link processInitMessage}
+   */
+  private initialize() {
+    void this.sendMessage({
+      partialMessage: { messageName: "init" },
+      respondedToBy: "initResponse",
+      sender: this,
+    });
+  }
+
+  /**
+   * Receives the {@link EmbedderInitMessage} sent by the embedding application,
+   * which is a series of payloads namespaced by service and message name.
+   * Calls the individual callbacks registered for each of these values.
+   *
+   * Useful for HTML blocks receiving messages which are sent on initialization by the app
+   * – other types of blocks can receive these messages synchronously via properties.
```

Comment:
Is there an example of this somewhere? I don't see how this is done. With the react integration, I don't see how an update to these properties triggers a re-render.

Reply:
Yes – this is really for blocks that can't receive props directly but must retrieve them. i.e. an arbitrary pile of HTML

Reply:
If our component will only have 1 (or few instances) of a hook, use object return. Please let me know if another kind of example or discussion would be useful